zerver/tests/test_subs.py

```diff
@@ -374,48 +375,86 @@ class StreamAdminTest(ZulipTestCase):
     def test_make_stream_public(self) -> None:
         user_profile = self.example_user('hamlet')
         self.login_user(user_profile)
-        self.make_stream('private_stream', invite_only=True)
+        self.make_stream('private_stream_1', invite_only=True)
+        self.make_stream('private_stream_2', invite_only=True)
 
         do_change_user_role(user_profile, UserProfile.ROLE_REALM_ADMINISTRATOR)
         params = {
-            'stream_name': ujson.dumps('private_stream'),
+            'stream_name': ujson.dumps('private_stream_1'),
             'is_private': ujson.dumps(False),
         }
-        stream_id = get_stream('private_stream', user_profile.realm).id
+        stream_id = get_stream('private_stream_1', user_profile.realm).id
         result = self.client_patch(f"/json/streams/{stream_id}", params)
         self.assert_json_error(result, 'Invalid stream id')
 
-        stream = self.subscribe(user_profile, 'private_stream')
+        stream = self.subscribe(user_profile, 'private_stream_1')
         self.assertFalse(stream.is_in_zephyr_realm)
 
         do_change_user_role(user_profile, UserProfile.ROLE_REALM_ADMINISTRATOR)
         params = {
-            'stream_name': ujson.dumps('private_stream'),
+            'stream_name': ujson.dumps('private_stream_1'),
             'is_private': ujson.dumps(False),
         }
         result = self.client_patch(f"/json/streams/{stream_id}", params)
         self.assert_json_success(result)
 
         realm = user_profile.realm
-        stream = get_stream('private_stream', realm)
+        stream = get_stream('private_stream_1', realm)
+        self.assertFalse(stream.invite_only)
+        self.assertTrue(stream.history_public_to_subscribers)
+
+        do_change_user_role(user_profile, UserProfile.ROLE_MEMBER)
+        params = {
+            'stream_name': ujson.dumps('private_stream_2'),
+            'is_private': ujson.dumps(False),
+        }
+        stream = self.subscribe(user_profile, 'private_stream_2')
+        result = self.client_patch(f"/json/streams/{stream.id}", params)
+        self.assert_json_error(result, "This action requires administrative rights.")
```

Comment:
Let's have this error message be formatted like the one for the realm administrators; can you make an exception class like  `OrganizationAdministratorRequired` called `StreamAdministratorRequired` with message "Must be an organization or stream administrator" for this?